src/libs/HttpUtils.ts

```diff
@@ -25,17 +26,40 @@ Onyx.connect({
 // We use the AbortController API to terminate pending request in `cancelPendingRequests`
 let cancellationController = new AbortController();
 
+/**
+ * The API commands that require the skew calculation
+ */
+const addSkewList = ['OpenReport', 'ReconnectApp', 'OpenApp'];
+
+/**
+ * Regex to get API command from the command
+ */
+const APICommandRegex = /[?&]command=([^&]+)/;
+
 /**
  * Send an HTTP request, and attempt to resolve the json response.
  * If there is a network error, we'll set the application offline.
  */
 function processHTTPRequest(url: string, method: RequestType = 'get', body: FormData | null = null, canCancel = true): Promise<Response> {
+    const startTime = new Date().valueOf();
     return fetch(url, {
         // We hook requests to the same Controller signal, so we can cancel them all at once
         signal: canCancel ? cancellationController.signal : undefined,
         method,
         body,
     })
+        .then((response) => {
+            const match = url.match(APICommandRegex)?.[1];
+            if (match && addSkewList.includes(match) && response.headers) {
+                const dateHeaderValue = response.headers.get('Date');
+                const serverTime = dateHeaderValue ? new Date(dateHeaderValue).valueOf() : new Date().valueOf();
+                const endTime = new Date().valueOf();
+                const latency = (endTime - startTime) / 2;
+                const skew = serverTime - startTime + latency;
+                NetworkActions.setTimeSkew(dateHeaderValue ? skew : 0);
+            }
+            return response;
+        })
```

Comment:
I think it would be great to leave a comment @waterim about why we are even doing this